File: Data Structures & Algorithms/two-integer-sum/submission-8.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
      const seen = new Map();
      for (let i = 0; i < nums.length; i++) {
         const num = nums[i];
         const needed = target - num;

         if (seen.has(needed)) {
            return [seen.get(needed), i];
         }

         seen.set(num, i);
      }

      return [];
    }
}

File: Data Structures & Algorithms/binary-search/submission-8.cs
public class Solution {
    public int Search(int[] nums, int target) {
        int left = 0;
        int right = nums.Length - 1;

        while (left <= right) {
          int mid = left + (right - left) / 2;
          if (nums[mid] == target) return mid;
          if (nums[left] < target) {
            left++;
          } else {
            right--;
          }
        }

        return -1;
    }
}
